src/Context.php

```diff
@@ -72,7 +72,13 @@ public function initialize(string $format, VisitorInterface $visitor, GraphNavig
         $this->metadataStack = new \SplStack();
 
         if (isset($this->attributes['groups'])) {
-            $this->addExclusionStrategy(new GroupsExclusionStrategy($this->attributes['groups']));
+            $strategy = new GroupsExclusionStrategy($this->attributes['groups']);
+
+            if (isset($this->attributes['inherited_groups']) && is_bool($this->attributes['inherited_groups'])) {
```

Comment:
As attributes are private, I guess we do not need to check if it is bool, as it should be enforced by the `enableInheritGroups()` method :)